### ISSUE TITLE
mustang doesn't work with lto

### ISSUE THREAD

**Issue by sunfishcode** [sunfishcode/mustang#72]
This reproduces with just adding
```toml
[profile.release]
lto = true
```
to the top-level Cargo.toml in the mustang repo, and running the `hello` example. It gets lots of undefined references to things like `core::panicking::assert_failed_inner`. See also https://github.com/sunfishcode/mustang/issues/22#issuecomment-983208986.

**Comment by HookedBehemoth** [comment]
This also happens with profile.dev for me. The error output is a different one though.
Similar to #138 I get this a bunch of times
```
inlinable function call in a function with debug info must have a !dbg location
  call void @_Unwind_Resume(ptr %393) #42
```
Release is as you described. Tried gcc, ld.lld and mold.

**Comment by whentze** [comment]
Is this still an issue? I tried to reproduce it on current `main` (38b4c917ada91fbea019b13c1a95cfb6197d4ad4) using `rustc 1.69.0-nightly (75a0be98f 2023-02-05)` and it just works. Maybe this got solved accidentally?

**Comment by sunfishcode** [comment]
It still reproduces for me with the latest nightly. Mustang currently depends on nightly, so if you're using stable it's likely you're using a non-mustang target.

**Comment by whentze** [comment]
I am using a nightly, just not the latest one, because that fails with ``unknown feature `atomic_mut_ptr` `` - looks like that API was superseded by [a newer, now stable one](https://github.com/rust-lang/rust/pull/108419).

I can now reproduce the issue - I was using a mustang target but had forgotten to add `--release` 🤦🏼

**Comment by sunfishcode** [comment]
Mustang still doesn't support LTO, however [origin](https://github.com/sunfishcode/origin) has now moved out to its own repo and become a standalone usable project, and it *does* support LTO. And [origin-studio](https://github.com/sunfishcode/origin-studio) provides a tiny std-like API for it. So if you want the Rust program startup and shutdown parts of mustang and don't need a full `std` or libc ABI compatibility, check out origin and/or origin-studio!

**Comment by sunfishcode** [comment]
This is now fixed in #239, which is now released in mustang 0.14.4. Mustang now supports LTO!
